### PR TITLE
Fix typos in setup and docs

### PR DESCRIPTION
**Pull request by keyserj**
Found a few issues when looking at the docs and running through the dev setup process. Happy to split these commits into separate PRs and/or remove some, just let me know.

The issues I ran into are these, and line up with the commits I made:
- default port for serving the api didn't match the port that the UI was sending requests to. updated the .example.env file
- setup commands require psql but that wasn't mentioned to be installed. I just added a mention to install psql - not sure if preference would be to add a specific install command like there is for node & yarn, but I also wasn't sure which command to add for this
- some setup commands assume they're running from the package directory. I updated to assume running from root. could also consider adding a `cd premiser-api` for that section, or maybe mention to run from that directory.
- the `create-user:local` command errored because zod validation failed. updated the fields to pass validation. I also searched `createUser` to see if there were tests to update, it looks like there aren't, but there are other `createUser` methods that use zod-invalid parameters (e.g. [acceptedTerms](https://github.com/Howdju/howdju/blob/4ec08f02a7bfef1eb754bb7f541c50f7c7717f9a/howdju-service-common/lib/services/UsersService.ts#L90) instead of [doesAcceptTerms](https://github.com/Howdju/howdju/blob/4ec08f02a7bfef1eb754bb7f541c50f7c7717f9a/howdju-common/lib/zodSchemas.ts#L1667)), maybe this is just unfinished zod migration?
- the docs look sweet. but the "edit this page" button directs to the `main` branch instead of `master`, which is invalid

Additionally, I ran `check:everything` locally and everything seems good except the `howdju-service-common` tests fail to create a test database (this occurs for every test in this package):
```
Started docker image 686746f6bede465550c8f08277e498382fc8a0a11e0646131326f8d4c4032817
Running: jest
 FAIL  lib/daos/PropositionsDao.test.ts (8.712 s)
  ● Console

    console.info
      Failed to create test DB howdju_cb10454714ebc85181e6c8171a589f4ed08add1b. Assuming it already exists and retrying a different name. Error: {"length":103,"name":"error","severity":"FATAL","code":"28P01","file":"auth.c","line":"333","routine":"auth_failed"}
```

I couldn't figure out how to get this going, and I tried looking in the test docs. It seems like the script change I made is only used for setup, so maybe these tests are irrelevant and/or GitHub Actions can run them for me? 😆

### CODE REVIEW

**Comment by carlgieringer** [comment]
> the `howdju-service-common` tests fail to create a test database

Yes, I think I am missing instructions to have a `config/test.env` file (it's used [here](https://github.com/Howdju/howdju/blob/4ec08f02a7bfef1eb754bb7f541c50f7c7717f9a/howdju-service-common/bin/test.sh#L7)). And actually I should probably just commit that to the repository so that no setup is required, but someone can modify it if they need to. The following contents should work:

```
DB_USER=postgres
DB_NAME=premiser
DB_PASSWORD=test-db-password
DB_HOST=localhost
DB_PORT=5433
LOG_LEVEL=silly
NODE_ENV=test
```

Regarding the Github actions, I need to figure out how to set them up for external PRs. They run automatically for PRs from branches in this repo, but I wasn't aware that they required additional config for PRs from other repos. I will spend a little time trying to figure it out, and otherwise we can just merge without those checks and deal with any breakages on master.

Thanks!

**Comment by keyserj** [comment]
I'm not sure if this is specifically the case, but I know that generally without configuration, actions need to be approved when PRs are made from forks https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks. It might look different in this case because workflow run requests are also deleted after 30 days? I can try pushing again to see if another workflow run request is made that you can approve

**Review by carlgieringer — APPROVED** [review]
LGTM. Thanks!

**Comment by carlgieringer** [comment]
The CI checks failed. I think it's because the GH action is triggered by `pull_request` which doesn't provide access to this repo's secrets or vars. I can update the action not to refer to those.

**Comment by carlgieringer** [comment]
Can you please rebase (now that #628 is merged) and see if the checks work? (I'm not so much concerned that your changes would break anything, but more interested in testing the process. I really appreciate your help!

File: docs/Development.md [hunk 1]
```diff
@@ -67,23 +67,23 @@ have `premiser_admin` for now for parity with the production database.
...
+cp ./config/example.env ./config/local.env
 ```
 
 Be sure to update `DB_PASSWORD` to be `premiser_api`'s password.
 
 ### Create a Howdju user
 
 ```shell
-yarn run create-user:local --email test@localhost --username=test_user
+yarn run create-user:local --email test@test.com --username=test_user
```
[the remark this answers is not in the document]
Comment (keyserj) [reply docs/Development.md]: I updated to use `test@test.test`

[the remark this answers is not in the document]
Comment (carlgieringer) [reply docs/Development.md]: Ah! Thanks for explaining. I think the reserved TLD sounds like a great option. Thanks for that suggestion!

**Comment by carlgieringer** [comment]
Great, the premerge checks passed. I will need to look more into the coverage check...it requires the GH token so that it can checkout the merge base, so I'm not sure it will work with pull_request.

**Comment by carlgieringer** [comment]
Thanks a ton, Joel!

Note to self: the DCO didn't seem to quote be recognized correctly by GH. The DCO check passed, and manually inspecting the commits shows that the correct sign off message is there, but Github still complained that "The base branch requires all commits to be signed".

Screenshot 2024-01-08 at 10 38 02 AM

**Pull request merged**